### PR TITLE
(#310) Adds Uninstall-ChocolateyPath and now respects semicolons in paths

### LINKED ISSUE #310: Add PowerShell helper function to manage removing paths from $PATH
There is a useful helper Install-ChocolateyPath which I use to add directories to the PATH variable. I would like to see an Uninstall-ChocolateyPath as well to cleanup those directories from the PATH variable at uninstall time.
I guess this would apply to quite a few other helpers which only have an "Install" part.

### PR DESCRIPTION
**Pull request by jnm2**
Closes #310 and #1019.

Discussion:
- https://github.com/chocolatey/choco/pull/1019#discussion_r164261454
- https://github.com/chocolatey/choco/pull/1019#issuecomment-360983979

### CODE REVIEW

**Comment by franklinyu** [comment]
I think this should also close #1019?

**Comment by jnm2** [comment]
1019 is a PR. Does it make sense to autoclose one PR from another?

**Comment by franklinyu** [comment]
Makes sense to me, because this PR supersedes #1019.

**Comment by jnm2** [comment]
@gep13 Is there anything I can do to help get this merged?

**Comment by jtcmedia** [comment]
Is this going to get merged? It would be extremely useful.

File: src/chocolatey.resources/helpers/functions/Uninstall-ChocolateyPath.ps1 [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  [parameter(Mandatory=$true, Position=0)][string] $pathToUninstall,
+  [parameter(Mandatory=$false, Position=1)][System.EnvironmentVariableTarget] $pathType = [System.EnvironmentVariableTarget]::User,
+  [parameter(ValueFromRemainingArguments = $true)][Object[]] $ignoredArguments
+)
+
+  Write-FunctionCallLogMessage -Invocation $MyInvocation -Parameters $PSBoundParameters
+
+  $paths = Parse-EnvPathList (Get-EnvironmentVariable -Name 'PATH' -Scope $pathType -PreserveVariables)
+  if ((IndexOf-EnvPath $paths $pathToUninstall) -eq -1) {
+    Write-Host "PATH environment variable does not have $pathToUninstall in it. Adding..."
```
Comment (jmcker) [inline src/chocolatey.resources/helpers/functions/Uninstall-ChocolateyPath.ps1:71]: This condition (and even the log) are identical to `Chocolatey-InstallPath`. I'm assuming it's supposed to be something more like:

`$removeIndex` is also not defined.

```suggestion
  $removeIndex = (IndexOf-EnvPath $paths $pathToUninstall)
  if ($removeIndex -ge 0) {
    Write-Host "Found $pathToUninstall in PATH environment variable. Removing..."
```

File: src/chocolatey.resources/helpers/functions/Uninstall-ChocolateyPath.ps1 [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    } else {
+      $paths = [System.Collections.ArrayList] $paths
+      $paths.RemoveAt($removeIndex)
+      Set-EnvironmentVariable -Name 'PATH' -Value $(Format-EnvPathList $paths) -Scope $pathType
+    }
+  }
+
+  # Make change immediately available
+  $paths = Parse-EnvPathList $env:PATH
+  if ((IndexOf-EnvPath $paths $pathToUninstall) -eq -1) {
```
Comment (jmcker) [inline src/chocolatey.resources/helpers/functions/Uninstall-ChocolateyPath.ps1:85]: Same deal as above.

```suggestion
  $removeIndex = (IndexOf-EnvPath $paths $pathToUninstall)
  if ($removeIndex -ge 0) {
```

**Review by jmcker — CHANGES_REQUESTED** [review]

**Comment by jmooring** [comment]
@gep13 Can you or someone else on your team have a look at this, please.

**Comment by pauby** [comment]
Please don't ping people on PR's. We are aware of all of the PR's and will look at them when we are able to.

**Comment by dazeeeed** [comment]
Amazing that this PR is still not merged in 2023, this is such a great feature.

**Comment by vexx32** [comment]
Apologies for the long wait on this PR. This work is being incorporated into a larger scope of work involving some rewrites that will be submitted soon, so for now we'll close this PR.

Thank you very much for the contribution, this was a really good place to start! 💜

**Pull request closed without merging**